### PR TITLE
Update CONTRIBUTING.md with development and contribution documentation

### LINKED ISSUE #19834: Unify development and contribution documentation
We have a few sources that help newcomers contribute to the project.

The following give higher level guidance on how to contribute:

* The [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md) file
* [The review and commit guidelines](../wiki/Review-and-Commit-guidelines)
* [This helpful document](https://docs.google.com/document/d/1PF2OkyRNQ7kgyLKK4aLqg78LSLjQxzTII6wg4yMTig8/edit#)

We also have the following code contribution guidelines that give more detailed help on how to build and write the code itself:

* The [README.md](../blob/master/README.md) file 
* [The Presto development guidelines](../wiki/Presto-Development-Guidelines)

We should consider either unifying the two classes of documentation into just two documents, or even consider flattening them into a single document (perhaps CONTRIBUTING.md).  The current situation is our documentation is spread out, which makes it difficult to ensure newcomers read all the appropriate documentation prior to contribution.  It also makes it difficult to know where to update the documentation when it needs to be updated.

### PR DESCRIPTION
**Pull request by VishnuSanal**
## Motivation and Context

fixes #19834

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [x] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [x] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [x] Adequate tests were added if applicable.
- [x] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by steveburnett** [comment]
I'm glad you took the initiative to get this moving! This looks good as a combination of the Presto [CONTRIBUTING.md](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md) with [README.md](https://github.com/prestodb/presto/blob/master/README.md). 

My assumption here is that the intent of #19834 is to reduce the number of pages, making it easier to find content. 

I also assume that if [CONTRIBUTING.md](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md) and [README.md](https://github.com/prestodb/presto/blob/master/README.md) are combined, that one of the two will go away. 

(If we keep both pages we run the risk of the same content on separate pages becoming out of sync and at least one being out of date.)

Let me know if my assumptions are incorrect!

Perhaps it might be better to combine the two files the other way, into README.md, as I believe README is more common a default starting page for a repo than CONTRIBUTING.

**Comment by steveburnett** [comment]
@VishnuSanal please sign the Contributor License Agreement to make contributions to Presto. For more information, see the [Contributor License Agreement](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#contributor-license-agreement-cla) topic in [CONTRIBUTING.md](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md).  

To sign a CLA, see the message above from the [linux-foundation-easycla](https://github.com/apps/linux-foundation-easycla) bot, and follow the link in [Please click here to be authorized](https://api.easycla.lfx.linuxfoundation.org/v2/repository-provider/github/sign/4238753/5349565/21450/#/?version=2).

**Comment by steveburnett** [comment]
@VishnuSanal , after you've signed the CLA, you can re-run the tests and pass the failing EasyCLA test. To do so, squash your commits and push, as described in the [Pull Release Checklist](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md#pull-request-checklist) topic in [CONTRIBUTING.md](https://github.com/prestodb/presto/blob/master/CONTRIBUTING.md).

**Comment by VishnuSanal** [comment]
hi, this is ready for review. I have moved the contribution guidelines exclusively to the CONTRIBUTING.md file & removed 'em from the project README. a link to CONTRIBUTING.md was given from the README to make it accessible. expecting a review & corrections if my assumptions are wrong, thanks!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -65,6 +148,30 @@ New committers are approved by majority vote of the TSC ([see TSC charter](https
...
+Fix OOM caused by foo in bar
+
+Foo was pack ratting ByteBuffers causing OOM.
+
+Cherry-pick of https://github.com/foo/bar/pull/123/commits/c2bf3b025ff456921c4c5ff4bcb814740e823061 (https://github.com/foo/bar/pull/123/commits/c2bf3b025ff456921c4c5ff4bcb814740e823061)
+
+Co-authored-by: Foo Bar <foo@bar.com>
+```
+
+- Example PR message
```
Comment (tdcmeehan) [inline CONTRIBUTING.md:164]: This is not accurate.  Please look at the PR template.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -65,6 +148,30 @@ New committers are approved by majority vote of the TSC ([see TSC charter](https
...
         * Use the body to explain what and why vs. how
+     
+- Example commit message
+
+```
+Fix OOM caused by foo in bar
+
+Foo was pack ratting ByteBuffers causing OOM.
+
+Cherry-pick of https://github.com/foo/bar/pull/123/commits/c2bf3b025ff456921c4c5ff4bcb814740e823061 (https://github.com/foo/bar/pull/123/commits/c2bf3b025ff456921c4c5ff4bcb814740e823061)
```
Comment (tdcmeehan) [inline CONTRIBUTING.md:159]:
```suggestion
Cherry-pick of c2bf3b0 (https://github.com/foo/bar/pull/123/commits/c2bf3b025ff456921c4c5ff4bcb814740e823061)
```

**Review by tdcmeehan — CHANGES_REQUESTED** [review]
Two minor comments--I think the examples need updating

**Comment by tdcmeehan** [comment]
Please squash commits--thanks!

**Comment by tdcmeehan** [comment]
Looks good, but please rewrite the commit message to conform to our guidelines.  Consider: `Improve CONTRIBUTING.md`

**Review by tdcmeehan — APPROVED** [review]

**Comment by tdcmeehan** [comment]
Thank you @VishnuSanal

**Pull request merged**